pkg/cvo/metrics.go

```diff
@@ -136,6 +143,21 @@ func RunMetrics(runContext context.Context, shutdownContext context.Context, lis
 		resultChannel <- asyncResult{name: "HTTPS server", error: err}
 	}()
 
+	certDir := filepath.Base(certFile)
+	origCertChecksum := checksumFile(certFile)
+	origKeyChecksum := checksumFile(keyFile)
+
+	// Set up and start the file watcher.
+	watcher, err := fsnotify.NewWatcher()
+	if watcher == nil || err != nil {
+		klog.Infof("!!!! Failed to create file watcher, cert/key rotation will be disabled  %v", err)
+	} else {
+		defer watcher.Close()
+		if err := watcher.Add(certDir); err != nil {
```

Comment:
Don't we need another of these for key-dir if the `keyFile` directory diverges from the `certFile` directory?